program/rust/src/processor/upd_price.rs

```diff
@@ -289,3 +345,35 @@ struct MessageBufferAccounts<'a, 'b: 'a> {
     oracle_auth_pda:     &'a AccountInfo<'b>,
     message_buffer_data: &'a AccountInfo<'b>,
 }
+
+#[cfg(test)]
+mod test {
+    use {
+        super::*,
+        crate::accounts::PriceComponent,
+        quickcheck_macros::quickcheck,
+        solana_program::pubkey::Pubkey,
+    };
+
+    /// Test the find_publisher_index method works with an unordered list of components.
+    #[quickcheck]
+    pub fn test_find_publisher_index_unordered_comp(comps: Vec<PriceComponent>) {
+        comps.iter().enumerate().for_each(|(idx, comp)| {
+            assert_eq!(find_publisher_index(&comps, &comp.pub_), Some(idx));
+        });
+
+        assert_eq!(find_publisher_index(&comps, &Pubkey::new_unique()), None);
```

Comment:
Is this technically wrong since `Pubkey::arbitrary()` can sometimes be equal to `Pubkey::new_unique()`

Reply:
Well this one is game of probabilities, the chances should be nearly zero.

Reply:
well it's not obvious actually. will fix it.